backend/src/main/java/com/klm/dev/exercises/devcase02/flight/FlightClient.java

```diff
@@ -0,0 +1,82 @@
+package com.klm.dev.exercises.devcase02.flight;
+
+import com.klm.dev.exercises.devcase02.weather.Weather;
+import com.klm.dev.exercises.devcase02.weather.WeatherClient;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+@Service
+@Configuration
+@PropertySource("classpath:Flight.properties")
+public class FlightClient {
+    @Autowired
+    private RestTemplate restTemplate;
+
+    private static final int NTHREDS = 20;
+
+    @Autowired
+    private WeatherClient weatherClient;
+
+    @Value("${flights.url}")
+    private String url;
+//Goes through all flights one by one, and calls the weather information on all locations
+// per flight
+    public Flight[] getFlights() {
+        Flight[] flights = restTemplate.getForObject(url, Flight[].class);
+        for (Flight flight: flights){
+           String[] cityName = flight.getRoute();
+            List<Weather> weathers = weatherClient.getWeathers(cityName);
+            Weather[] weather = new Weather[weathers.size()];
+            weather = weathers.toArray(weather);
+            flight.setWeather(weather);
+
+        }
+        return flights;
+    }
+    public Flight[] getFlights2() {
+        //First the method gets all flights. Than it iterates through the flights, and the
+        // routs of flights. It adds the airport code-name of the route to a String ArrayList
+        // if it is not yet in the list.
+        //It converts the ArrayList to an array, and makes a paralell call to the weather API
+        // with all locations at once.
+
+        Flight[] flights = restTemplate.getForObject(url, Flight[].class);
+        Set<String> locations = new HashSet<>();
+        for (Flight flight: flights){
+            String[] cityName = flight.getRoute();
+            for (String city : flight.getRoute()){
+                    locations.add(city);
+            }
+        }
+        String[] location = new String[locations.size()];
+        List<Weather> weathers = weatherClient.getWeathers(locations.toArray(location));
```

Comment:
when you retrieved the list convert it to a map so you can remove the unnecessary loop when you retrieve the correct weather for the station